CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/apis/OpenAiTests.cs

```diff
@@ -162,6 +162,47 @@ public class OpenAiTests {
             return emotionalChatResponse;
         }
 
+
+        [Fact]
+        public async Task ExampleUsage5_ImageToText() {
+            var openAi = new OpenAi(await IoC.inject.GetAppSecrets().GetSecret("OpenAiKey"));
+
+            var prompt = "A picture of a dog";
+            var result = await openAi.TextToImage(new OpenAi.Image.Request() { prompt = prompt });
+            var url = result.data.First().url;
+            Assert.NotEmpty(url);
+
+            var messages = new List<VisionGpt.Line>() {
+                new VisionGpt.Line(VisionGpt.Role.user, content: getFormattedContent(url))
+            };
+            var request = new VisionGpt.Request(messages);
+
+            var resultImageToText = await openAi.ImageToText(request);
+            var guessedContent = resultImageToText.choices[0].message.content.ToString();
+
+            Assert.NotEmpty(guessedContent);
+            Log.d("result id is: " + guessedContent);
+            Assert.Contains("dog", guessedContent);
+        }
+        private List<Dictionary<string, object>> getFormattedContent(string url) {
```

Comment:
This getFormattedContent method here should not be part of the tests right? This has to be part of the OpenAI class (except the concrete instructions in line 192)

Reply:
61e7fcf278e66d1dd4c3f69d38f881b71a397c9a